### ISSUE TITLE
[1.12.2] TexFix sometimes causes client crashes, because of Logistics Pipes

### ISSUE THREAD

**Issue by fkrisi11** [RS485/LogisticsPipes#1423]
When you go near Logistics Pipes in the world, sometimes the game crashes with the following error:
https://hastebin.com/olakipisox.rb

Is there something you can do about it, or is this an issue on the TexFix side?

**Comment by bziemons** [comment]
I can't find the source code for TexFix on GitHub or on their Curseforge site, so I don't even know what we could do or what the problem is. Sorry about that.

**Comment by fkrisi11** [comment]
I asked the dev of the TexFix mod to either release the source code or to try to fix it himself, but.. we'll have to see if he replies

**Comment by Speiger** [comment]
Eh the error is that a "null entry" gets added into the queue of my unload list which should be impossible unless you are loading textures on a seperate thread. That could be the only reason why.

Edit: With loading on a seperate thread i mean accessing the texturedata in a thread that isnt the client thread

**Comment by fkrisi11** [comment]
@bziemons Have you seen this?
I'm sorry if you have, but I do not know if you get a notification on closed issues if someone comments.

**Comment by Speiger** [comment]
@fkrisi11 yes I answered him also on curseforge

**Comment by bziemons** [comment]
> I'm sorry if you have, but I do not know if you get a notification on closed issues if someone comments.

Yes, sorry about the silence. We'll have to investigate. I am certain we are not using a separate thread to load textures, but we do have texture errors and that may be a part of the problem.

**Comment by fkrisi11** [comment]
It's all good. Thank you for your input. I hope it is something that is rather easy to track down ^^

**Comment by Speiger** [comment]
@bziemons can you provide me with your texture loading system so i can investigate a bit and maybe see what the issue is.

And to provide you with a bit of info:
Whenever the internal List of int[] is accessed (yes i replace the list implementation) the tool issues a texture load request. But only AFTER all resources were reloaded.

**Comment by bziemons** [comment]
@Speiger I can't, because I need to look for it myself. I cannot remember developing on that part ever.
